falcon/util/structures.py

```diff
@@ -141,6 +142,25 @@ class Context:
     True
     """
 
+    # NOTE(vytas): Define synthetic attr access methods (under TYPE_CHECKING)
+    #   merely to let mypy know this is a namespace object.
+    if TYPE_CHECKING:
+
+        def __getattr__(self, name: str) -> Any:
```

Comment:
there is no need to provide an implementation here.

Just define the function:
```py
def __getattr__(self, name: str) -> Any:...
def __setattr__(self, name: str, value: Any) -> None:...
def __delattr__(self, name: str) -> None:...
```

the type checker does not execute the code and `TYPE_CHECKING` is never true so the implementation has no reason to be there

Reply:
Ah, right :man_facepalming: I was first prototyping without `TYPE_CHECKING` and then never removed the impls.